python/az/aro/azext_aro/_rbac.py

```diff
@@ -37,6 +36,45 @@ def _create_role_assignment(auth_client, resource, params):
             logger.warning("%s; retry %d of %d", ex, retries, max_retries)
 
 
+def assign_contributor_to_resources(cmd, vnet, subnets, sp_id, fail=True):
+    # Assume that the role exists if we encounter an error
+    has_role = True
+
+    # Assign a Network Contributor role assignment to the service principal
+    # Return an error/exception if unable to create the role assignment, will
+    # skip if cannot view the current role assignments for a service principal on a resource
+    try:
+        has_role = has_network_contributor(cmd.cli_ctx, vnet, sp_id)
+    except CloudError as e:
```

Comment:
Since all of the blocks below check for the same exception type, we can wrap this in a single `try` block and DRY this code up.

Reply:
Refactored it back into the original methods and add an optional param to raise an exception if we fail to list the role assignments on a resource.